src/scribe_data/checkquery.py

```diff
@@ -0,0 +1,442 @@
+#!/usr/bin/env python3
+
+"""Command line tool for testing SPARQl queries against an endpoint."""
+
+import argparse
+import math
+import os
+import subprocess
+import sys
+import time
+import urllib.request
+from dataclasses import dataclass
+from http import HTTPStatus
+from pathlib import Path
+from typing import Optional
+from urllib.error import HTTPError
+
+import SPARQLWrapper as SPARQL
+from SPARQLWrapper import SPARQLExceptions
+from tqdm.auto import tqdm
+
+
+EXIT_SUCCESS = 0
+EXIT_FAILURE = 1
+EXIT_CLI_ERROR = 2
+
+PROJECT_ROOT = "Scribe-Data"
+
+
+@dataclass(repr=False, frozen=True)
+class QueryFile:
+    """Holds a reference to a file containing a SPARQL query."""
+
+    path: Path
+
+    def load(self, limit: int) -> str:
+        """Load the SPARQL query from 'path' into a string.
+
+        Args:
+            limit (int): the maximum number of results a query should return.
+
+        Returns:
+            str: the SPARQL query.
+        """
+        with open(self.path, encoding="utf-8") as in_stream:
+            return f"{in_stream.read()}\nLIMIT {limit}\n"
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}(path={self.path})"
+
+
+class QueryExecutionException(Exception):
+    """Raised when execution of a query fails."""
+
+    def __init__(self, message: str, query: QueryFile) -> None:
+        """
+        Args:
+            message (str): why the query failed.
+            query (QueryFile): the query that failed.
+        """
+        self.message = message
+        self.query = query
+        super().__init__(self.message)
+
+    def __str__(self) -> str:
+        return f"{self.query.path} : {self.message}"
+
+
+def ping(url: str, timeout: int) -> bool:
+    """Test if a URL is reachable.
+
+    Args:
+        url (str): the URL to test.
+        timeout (int): the maximum number of seconds to wait for a reply.
+
+    Returns:
+        bool: True if connectivity established. False otherwise.
+    """
+    try:
+        with urllib.request.urlopen(url, timeout=timeout) as response:
+            return response.getcode() == HTTPStatus.OK
+    except (HTTPError, Exception) as err:
+        print(f"{type(err).__name__}: {str(err)}", file=sys.stderr)
+
+    return False
+
+
+def all_queries() -> list[QueryFile]:
+    """All the SPARQL queries in, and below, 'Scribe-Data/'.
+
+    Returns:
+        list[QueryFile]: the SPARQL query files.
+    """
+    parts = Path(__file__).resolve().parts
+    prj_root_idx = parts.index(PROJECT_ROOT)
+    prj_root = str(Path(*parts[: prj_root_idx + 1]))
+
+    queries: list[QueryFile] = []
+
+    for root, _, fnames in os.walk(prj_root):
+        for fname in fnames:
+            fpath = Path(root, fname)
+            if fpath.suffix == ".sparql":
+                queries.append(QueryFile(fpath))
+
+    return queries
+
+
+def changed_queries() -> Optional[list[QueryFile]]:
+    """Find all the SPARQL queries that have changed.
```

Comment:
This is so cool! And also adds so much value to this process! Being able to run all of  them on the off week to make sure that we can split them if need be, plus using `changed_queries()` whenever there's an edit via a PR!